q2_diversity_lib/plugin_setup.py

```diff
@@ -27,23 +28,88 @@
+                         Properties('phylogenetic', 'qualitative')])],
     input_descriptions={
         'table': 'The feature table containing the samples for which Faith\'s '
-                 'phylogenetic diversity should be computed.',
+                 'phylogenetic diversity should be computed. Table data will '
+                 'be converted to presence/absence.',
         'phylogeny': 'Phylogenetic tree containing tip identifiers that '
                      'correspond to the feature identifiers in the table. '
                      'This tree can contain tip ids that are not present in '
                      'the table, but all feature ids in the table must be '
                      'present in this tree.'},
     parameter_descriptions=None,
-    output_descriptions={'faith_pd': 'Vector containing per-sample values for '
-                                     'Faith\'s Phylogenetic Diversity.'},
+    output_descriptions={'vector': 'Vector containing per-sample values for '
+                                   'Faith\'s Phylogenetic Diversity.'},
     name='Faith\'s Phylogenetic Diversity',
     description='Computes Faith\'s Phylogenetic Diversity for all samples in '
                 'a feature table.',
     citations=[citations['faith1992conservation']]
 )
+
+plugin.methods.register_function(
+    function=q2_diversity_lib.observed_features,
+    inputs={'table': FeatureTable[Frequency | RelativeFrequency
+            | PresenceAbsence]},
+    parameters=None,
+    outputs=[('vector',
+             SampleData[AlphaDiversity % Properties('non-phylogenetic',
+                                                    'qualitative')])],
+    input_descriptions={'table': 'The feature table containing the samples '
+                        'for which the number of observed features should be '
+                        'calculated.'},
+    parameter_descriptions=None,
+    output_descriptions={'vector': 'Vector containing per-sample counts of '
+                                   'observed features.'},
+    name='Observed Features',
+    description='Compute the number of observed features for each sample in a '
+                'feature table',
+    citations=[citations['sokal1963principles']]
+)
+
+plugin.methods.register_function(
+    function=q2_diversity_lib.pielou_evenness,
+    inputs={'table': FeatureTable[Frequency | RelativeFrequency]},
+    parameters={'drop_nans': Bool},
+    outputs=[('vector',
+             SampleData[AlphaDiversity % Properties('non-phylogenetic',
+                                                    'quantitative')])],
+    input_descriptions={'table': 'The feature table containing the samples '
+                        'for which Pielou\'s evenness should be computed.'},
+    parameter_descriptions={'drop_nans': 'Samples with fewer than two observed'
+                            ' features produce undefined (NaN) values. '
+                            'Passing \'True\' drops these samples from the '
```

Comment:
This message encodes a bit of interface-specific logic into it, I think you should be able to re-write in a way that eschews the Artifact API.